### ISSUE TITLE
Bundling issues with cy-verify-downloads and cypress-delete-downloads-folder - plugins that use node built in packages? 

### ISSUE THREAD

**Issue by DavidAtImpactCubed** [badeball/cypress-cucumber-preprocessor#952]
I'm just getting into Cypress testing, and I'm a big fan of cucumber where I have used it in other test frameworks, so first of big thanks for this plugin, and apologies if my lack of knowledge is leading me to make stupid mistakes!

### Current behavior

I am porting existing tests that use the following plugins to cucumber:
https://github.com/elaichenkov/cy-verify-downloads
https://github.com/elaichenkov/cypress-delete-downloads-folder

These plugins worked as expected when using plain cypress with Typescript. Now with the cucumber pre-processor, they cause errors relating to targeting node. For example, using cy-verify-downloads causes:
`The package "fs" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.`
and
`The package "crypto" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.`

**TLDR: Is there a way I can use these packages with cypress-cucumber-preprocessor?**

This package is configured as follows:
In e2e.ts:
`require('cy-verify-downloads').addCustomCommand();`
In cypress.config.ts:
`const { verifyDownloadTasks } = require('cy-verify-downloads');`
... and in setupNodeEvents:
`on('task', verifyDownloadTasks);`

It is then used as follows:
```
Cypress.Commands.add('checkFileExists', (filename: string) => {
  cy.verifyDownload(filename);
});
```

Here is what I have tried to resolve the issue (none of these ideas worked):

**1. I tried excluding the package from the bundler, like this:**
```
      on(
        "file:preprocessor",
        createBundler({
          external: ['cy-verify-downloads'],
          plugins: [createEsbuildPlugin(config)],
        })
      );
```
Which caused the following error to surface in the test runner:
`Dynamic require of "cy-verify-downloads" is not supported`

**2. So, in addition to the exclude above, I tried to use the cy.task function to load the package dynamically at runtime:
In the tests:**
I removed the addCustomCommand from e2e.ts.
```
Cypress.Commands.add('verifyDownloads', () => {
  return cy.task('loadPackage', 'cy-verify-downloads').then((cyVerifyDownloads) => {
    cyVerifyDownloads.verifyDownloads();
  });
});
```
In setupNodeEvents:
```
      on('task', {
        loadPackage(packageName) {
          const packagePath = path.resolve('node_modules', packageName);
          const moduleExports = execSync(`node -p "require('${packagePath}')"`);
          return JSON.parse(moduleExports.toString());
        },
      });
```
This gave the following error:
```
cy.task('loadPackage') failed with the following error:

> Command failed: node -p "require('C:\[path-to-my-repository]\node_modules\cy-verify-downloads')"
node:internal/modules/cjs/loader:1063
  throw err;
  ^
Error: Cannot find module 'C:\[path-to-my-repository]\node_modules\cy-verify-downloads'
```
I have checked and C:\[path-to-my-repository]\node_modules\cy-verify-downloads exists. 

At this stage, I'm really not sure what to do (or whether what I've already tried is even relevant). Am I just unlucky in that the first two npm packages I have tried to use are incompatible?

My apologies if the issue lies with my understanding of how this should (or should not) work. 

### Desired behavior

Some way (any way!) to use these packages with the cucumber preprocessor. 

### Test code to reproduce

I haven't created one, but can if it would help.

### Versions

* **Cypress version**: 12.6.0
* **Preprocessor version**: 15.1.4
* **Node version**: 19.6.0

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by DavidAtImpactCubed** [comment]
I the issue here may simply be because we have a single npm package that contains both cypress specific code and node specific code. One to add the custom commands, and the other to be added to the NodeEvents pipeline. The issue crops up because the package as a whole is either bundled (causing an error due to the node specific code) or not bundled (in which case the custom commands aren't added?)

I'm not really sure to be honest, but I have a functioning workaround.

I re-implemented the libraries to assert on the file system myself. This allows me to keep the node stuff within the pipeline and out of the bundler.

For anyone who might find this issue, here are my workarounds:

```
    async setupNodeEvents(
      on: Cypress.PluginEvents,
      config: Cypress.PluginConfigOptions
    ): Promise<Cypress.PluginConfigOptions> {
      await addCucumberPreprocessorPlugin(on, config);

      on("file:preprocessor", createBundler({ plugins: [createEsbuildPlugin(config)], }));

      on('task', { fileExists });
      on('task', { deleteFolder });

      return config;
    },
```

```
function fileExists(filename) {
  return new Promise((resolve, reject) => {
    var dir = path.dirname(filename);
    if (!existsSync(dir)) {
      mkdirSync(dir);
    }

    var timer = setTimeout(function () {
      watcher.close();
      reject(new Error(`${filename} did not exist and was not created during the timeout.`));
    }, 40000);

    var watcher = watch(dir, function (eventType, eventFileName) {
      if (existsSync(filename)) {
        clearTimeout(timer);
        watcher.close();
        resolve(true);
      }
    });
    resolve(true);
  });
};

function deleteFolder(folderName) {
  console.log('deleting folder %s', folderName);

  return new Promise((resolve, reject) => {
    if (existsSync(folderName)) {
      rmdir(folderName, { maxRetries: 10, recursive: true }, (err) => {
        if (err) {
          console.error(err);
          return reject(err);
        }
      });
    }

    resolve(null);
  });
};
```

```
export const deleteDownloadsFolder = () => {
  const downloadsFolder = Cypress.config('downloadsFolder');
  cy.task('deleteFolder', downloadsFolder);
};

export const checkDownloadExists = (filename: string) => {
  const downloadsFolder = Cypress.config('downloadsFolder');
  cy.task('fileExists', `${downloadsFolder}\\${filename}`);
};
```

**Comment by badeball** [comment]
https://github.com/elaichenkov/cy-verify-downloads/issues/83
